### PR TITLE
Add Create Collection to dropdown

### PR DESCRIPTION
**Pull request by lusol**
Made the collection modal into a component for reusability purposes.
Can now create collections from drop downs.
screen shot 2018-06-20 at 12 17 23 pm
Note the Create button at the end of dropdown

issue: https://github.com/screwdriver-cd/screwdriver/issues/727

### CODE REVIEW

**Comment by Filbird** [comment]
PR title feels a bit bulky and is a little hard to understand. Also please include the issue. I assume this is the one:
https://github.com/screwdriver-cd/screwdriver/issues/727

Can you shorten it to something like "feat(727): Add 'CREATE' to collection dropdown"  

In the commit add the issue in semantic version form as well.

**Comment by tkyi** [comment]
Test failed:
```
12:26:07   not ok 69 Chrome 67.0 - ESLint | app: components/search-list/component.js
12:26:07       ---
12:26:07           actual: >
12:26:07               false
12:26:07           expected: >
12:26:07               true
12:26:07           stack: >
12:26:07                   at Object.<anonymous> (http://localhost:7357/assets/tests.js:1055:12)
12:26:07                   at runTest (http://localhost:7357/assets/test-support.js:4108:30)
12:26:07                   at Test.run (http://localhost:7357/assets/test-support.js:4094:6)
12:26:07                   at http://localhost:7357/assets/test-support.js:4306:12
12:26:07                   at Object.advance (http://localhost:7357/assets/test-support.js:3746:26)
12:26:07                   at begin (http://localhost:7357/assets/test-support.js:5695:20)
12:26:07           message: >
12:26:07               components/search-list/component.js should pass ESLint
12:26:07               
12:26:07               64:48 - Unexpected unnamed function. (func-names)
12:26:07               67:27 - Unexpected unnamed function. (func-names)
12:26:07               67:27 - Unexpected function expression. (prefer-arrow-callback)
12:26:07           Log: |
12:26:07       ...
```

File: app/components/search-list/component.js [hunk 1]
```diff
@@ -58,7 +60,28 @@ export default Component.extend({
       });
     }
   }),
+  helper() {
+    $('ul.dropdown.dropdown-menu').on('click', function () {
```
Comment (jithin1987) [inline app/components/search-list/component.js]: These two event handlers are getting created during each call to `helper()`. They should be created once and cleaned up when this component dies.

File: app/components/search-list/component.js [hunk 2]
```diff
@@ -58,7 +60,28 @@ export default Component.extend({
       });
     }
   }),
+  helper() {
```
Comment (jithin1987) [inline app/components/search-list/component.js]: need a better design for this helper. we can't keep attaching events to nodes on a function call without any cleanup. Perhaps there is an ember native way on what this is trying to do without adding custom event handlers.

File: app/components/search-list/component.js [hunk 3]
```diff
@@ -58,7 +60,28 @@ export default Component.extend({
...
+        $('ul.dropdown.dropdown-menu').parent().removeClass('open');
+      }
+    });
+  },
   actions: {
+    openModal() {
+      this.helper();
+      this.set('showModal', true);
+    },
+    addNewCollectionHelper() {
```
Comment (jithin1987) [inline app/components/search-list/component.js]: is this action used ?

File: tests/integration/components/collections-flyout/component-test.js [hunk 4]
```diff
@@ -221,21 +173,36 @@ test('it fails to create a collection', function (assert) {
...
-  this.set('showModal', false);
-
-  this.render(hbs`{{collections-flyout collections=collections showModal=showModal}}`);
-
-  $('.new').click();
-  assert.ok(this.get('showModal'));
-  $('.collection-form__cancel').click();
-  assert.notOk(this.get('showModal'));
-});
+// test('it cancels creation of a collection', function (assert) {
```
Comment (jithin1987) [inline tests/integration/components/collections-flyout/component-test.js]: why is this commented out ?

**Pull request merged**